src/Microsoft.Azure.WebJobs.Host/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace System.Collections.Generic
+{
+    internal static class DictionaryExtensions
+    {
+        public static T GetValueOrNull<TKey, TValue, T>(this IDictionary<TKey, TValue> dictionary, TKey key) where T : class
```

Comment:
That's still possible -- the Dictionary constructor is where you define `StringComparer`. So you can create a dictionary that way and use it with this method.